File: src/dotnet/Src/ButterflyNet.Parser/Scopes/MacroScope.cs
namespace ButterflyNet.Parser.Scopes {
	public class MacroScope : InlineScope {
		public MacroScope(IButterflyMacro macro) {
			Macro = macro;
		}

		public IButterflyMacro Macro { get; private set; }
		public override bool CanNestText { get { return false; } }

		public override void Open(ButterflyAnalyzer analyzer) {
			analyzer.OpenMacro(Macro);
		}

		public override void Close(ButterflyAnalyzer analyzer) {
			analyzer.CloseMacro(Macro);
		}
	}
}

File: src/js/src/scopes/MacroScope.js
function MacroScope(macro) {
	this.type = ScopeTypeCache.macro;
	this.level = ScopeLevel.inline;
	this.macro = macro;
	
	this.open = function(analyzer) {
		analyzer.openMacro(this.macro);
	};
	this.close = function(analyzer) {
		analyzer.closeMacro(this.macro);
	};
}

extend(Scope, MacroScope);